cmake/OpenCVModule.cmake

```diff
@@ -877,7 +877,9 @@ macro(_ocv_create_module)
 
   ocv_compiler_optimization_process_sources(OPENCV_MODULE_${the_module}_SOURCES OPENCV_MODULE_${the_module}_DEPS_EXT ${the_module})
   set(__module_headers ${OPENCV_MODULE_${the_module}_HEADERS})
-  list(SORT __module_headers)  # fix headers order, useful for bindings
+  if(__module_headers)
+    list(SORT __module_headers)  # fix headers order, useful for bindings
+  endif()
```

Comment:
BTW, Looks like this problem is related to old CMake only (reproduced with CMake 2.8.12, CMake 3.18 is OK):

Alternative fix of this problem is adding quotes `""` in the `set()` call:
```
set(__module_headers "${OPENCV_MODULE_${the_module}_HEADERS}")
list(SORT __module_headers)
```